docs/gitrepo-structure.md

```diff
@@ -53,6 +53,17 @@ defaultNamespace: default
 # Default: ""
 namespace: default
 
+# Specify a deployment schedule to deploy bundle during that window.
+# In this example we will deploy bundles every monday from 4pm to 5pm.
+# If a helm timeout is specified in the helm structure below, it will be considered in the schedule evaluation.
```

Comment:
I'm not too familiar with the helm code. I'm a bit concerned this will not be obvious to users, maybe we need to log that the "job will not be executed, because its timeout is too large for the schedule window".

But is it really necessary to take the helm timeout into account? As far as I  understand helm tasks are never cancelled by, so we rely on the helm library to honor the timeout value. If timeout is 0, `wait` is not used, are we certain helm will finish immediately in that case? I feel like it must be possible to make it take very long, by using hooks or lookups maybe?
In any case the actual helm deployment going on in the cluster might take longer. So even when you take the tiemout into consideration, there is no guarantee the cluster is in a ready state after the schedule window?

Reply:
Hm, okay. Best effort is fine with me.

However, I was looking at the helm code: https://github.com/rancher/fleet/blob/5a141a6b18f0e778b1ba1c248d76f9b03c0bf06c/pkg/helmdeployer/deployer.go#L328-L333 and I think the timeout defaults to 0 seconds. The helm package then uses that timeout [only to wait for hooks](https://github.com/rancher/helm/blob/release-v3.6.0/pkg/action/hooks.go#L83) and interprets 0 as forever (ContextWithOptionalTimeout).

So, if you tested your hook chart without a timeout, I think it would run for 350s.

Reply:
Indeed I see what you mean, specifying --timeout=0 run until it completes. It looks like fleet is setting 0 implicitly instead of the default 300 sec of helm, thus it will run forever and ever if the user do not specify a timeout right?